fineract-provider/src/main/java/org/apache/fineract/infrastructure/documentmanagement/api/ImagesApiResource.java

```diff
@@ -165,8 +165,13 @@ public Response retrieveImage(@PathParam("entity") final String entityName, @Pat
 
         try {
             byte[] resizedImageBytes = resizedImage.getByteSource().read();
-            final String clientImageAsBase64Text = imageDataURISuffix + Base64.getMimeEncoder().encodeToString(resizedImageBytes);
-            return Response.ok(clientImageAsBase64Text, MediaType.TEXT_PLAIN_TYPE).build();
+
+            if(resizedImageBytes != null){
```

Comment:
Not sure if this is right to handle this way.

I assume the resizedImageBytes is only null when there wasn't any image to resize. So I guess if we handle that use-case this way, the octet-stream branch might still fail.

I'd rather ask the question, is it reasonable to deal with these type of images?

Maybe at the beginning of this API, we shall simply put a check for this case and return an HTTP 400 as retrieving an image that doesn't exist (to some sense) is not necessarily something that makes sense for me, at least now.

Do you mind explaining the exact use-case for this extra logic?

Reply:
@galovics Hey! Thanks for jumping on this review - on a Saturday no less. So, the context here is that @temi-ro is a student who I'm teaching how to FLOSS and GitHub... we picked FINERACT-1224 as an example, and this is his very first contribution. We haven't gotten to actually running Fineract, and uploading images, and fully reproducing the issue. This would take more time than we have together today. The question is if this PR as-is (once it passed the build, of course) could be acceptable to you as a first step to at least avoid the NPE and clearer diagnostic log, as a first step? Still better than the NPE, IMHO. (I may follow-up with @temi-ro to further improve this case when I show him how to actually run everything etc.)

Reply:
Sure thing. Thanks for the context @vorburger.